mta/tests/migration/application_inventory/test_analysis.py

```diff
@@ -1,8 +1,21 @@
 import pytest
 from wait_for import wait_for
 
+from mta.fixtures.application_inventory import json_application
+from swagger_client.models.api_task import ApiTask
 
+
+@pytest.mark.parametrize("application", json_application(), indirect=True)
 @pytest.mark.analysis
-def test_source_analysis(tasks, get_api):
-    for task in tasks:
-        wait_for(lambda: get_api.tasks_id_get(str(task.id)).state == "Succeeded", delay=5, timeout=180)
+def test_analysis(application, analysis_item, create_api, get_api, update_api):
+    app = application
+    analysis_info = analysis_item
+    if app.name != analysis_info["app_name"]:
+        pytest.skip("The analysis item should not run using the current app. Skipping test")
```

Comment:
Mmm. so why use this application is this test in the 1st place?

Reply:
Because among the analysis entries provided nin analysis.json - there is **at least** one analysis details for each created app